hosts/galileo/__init__.py

```diff
@@ -0,0 +1,395 @@
+        [cspyce.furnsh(spkfile) for spkfile in glob.glob(kdir + 'SPK/*.bsp')]
+
+
+        return
+
+## TODO:
+        # Find the range of months needed
+        m1 = int((t0 - Galileo.TDB0) // Galileo.DTDB)
+        m2 = int((t1 - Galileo.TDB0) // Galileo.DTDB) + 1
+
+        m1 = max(m1, 0)         # ignore time limits outside mission duration
+        m2 = min(m2, Galileo.MONTHS - 1)
+
+        # Load any months not already loaded
+        for m in range(m1, m2+1):
+          if not loaded[m]:
+            for kernel in lists[m]:
+                filespec = kernel.filespec
+                if filespec not in kernel_dict:
+                    spicedb.furnish_kernels([kernel])
+                    kernel_dict[filespec] = kernel
+                loaded[m] = True
+
+    ############################################################################
+    # Initialize the kernel lists
+    ############################################################################
+
+    @staticmethod
+    def initialize_kernels(kernels, lists):
+        """After initialization, lists[m] is a the KernelInfo objects needed
+        within the specified month.
+        """
+        for i in range(Galileo.MONTHS):
+            lists[i] = []
+
+        for kernel in kernels:
+
+            # Find the range of months applicable, extended by 12 hours
+            t0 = cspyce.str2et(kernel.start_time) - Galileo.SLOP
+            t1 = cspyce.str2et(kernel.stop_time)  + Galileo.SLOP
+
+            m1 = int((t0 - Galileo.TDB0) // Galileo.DTDB)
+            m2 = int((t1 - Galileo.TDB0) // Galileo.DTDB) + 1
+
+            m1 = max(m1, 0)     # ignore time limits outside mission duration
+            m2 = min(m2, Galileo.MONTHS - 1)
+
+            # Add this kernel to each month's list
+            for m in range(m1, m2+1):
+                lists[m] += [kernel]
+
+    ############################################################################
+    # Routines for managing the loading other kernels
+    ############################################################################
+
+    @staticmethod
+    def load_instruments(instruments=[], asof=None):
+        """Load the SPICE kernels and defines the basic paths and frames for
+        the Galileo mission.
+
+        It is generally only to be called once.
+
+        Input:
+            instruments an optional list of instrument names for which to load
+                        frames kernels. The frames for ISS, VIMS, UVIS, and CIRS
+                        are always loaded.
+
+            asof        if this specifies a date or date-time in ISO format,
+                        then only kernels that existed before the specified date
+                        are used. Otherwise, the most recent versions are always
+                        loaded.
+        """
+
+        # Load the default instruments on the first pass
+        if Galileo.loaded_instruments == []:
+           instruments += ['SSI']
```

Comment:
append?

Reply:
same as above comment.  Cassini does it this way.  Does it matter?